tensorflow/tools/ci_build/builds/pip.sh

```diff
@@ -0,0 +1,287 @@
+  echo "Found local Python library directory at: ${LIB_PYTHON_DIR}"
+fi
+
+PACKAGES_DIR=`ls -d ${LIB_PYTHON_DIR}/*-packages | head -1`
+
+echo "Copying some source directories that are required by tests but are "\
+"not included in install to Python packages directory: ${PACKAGES_DIR}"
+
+# tensorflow.python.tools
+rm -rf ${PACKAGES_DIR}/tensorflow/python/tools
+cp -r tensorflow/python/tools \
+      ${PACKAGES_DIR}/tensorflow/python/tools
+touch ${PACKAGES_DIR}/tensorflow/python/tools/__init__.py  # Make module visible
+
+echo "Copying additional files required by tests to working directory "\
+"for test: ${PY_TEST_DIR}"
+
+# Image files required by some tests, e.g., images_ops_test.py
+mkdir -p ${PY_TEST_DIR}/tensorflow/core/lib
+rm -rf ${PY_TEST_DIR}/tensorflow/core/lib/jpeg
+cp -r tensorflow/core/lib/jpeg ${PY_TEST_DIR}/tensorflow/core/lib
+rm -rf ${PY_TEST_DIR}/tensorflow/core/lib/png
+cp -r tensorflow/core/lib/png ${PY_TEST_DIR}/tensorflow/core/lib
+
+# Run tests
+DIR0=`pwd`
+ALL_PY_TESTS=`find tensorflow/python -name "*_test.py"`
+PY_TEST_COUNT=`echo ${ALL_PY_TESTS} | wc -w`
+
+if [[ ${PY_TEST_COUNT} -eq 0 ]]; then
+  echo "ERROR: Cannot find any tensorflow Python unit tests to run on install"
+  exit 1
+fi
+
+# Iterate through all the Python unit test files using the installation
+COUNTER=0
+PASS_COUNTER=0
+FAIL_COUNTER=0
+SKIP_COUNTER=0
+FAILED_TESTS=""
+FAILED_TEST_LOGS=""
+
+for TEST_FILE_PATH in ${ALL_PY_TESTS}; do
+  ((COUNTER++))
+
+  PROG_STR="(${COUNTER} / ${PY_TEST_COUNT})"
+
+  # If PY_TEST_WHITELIST is not empty, only the white-listed tests will be run
+  if [[ ! -z ${PY_TEST_WHITELIST} ]] && \
+     [[ ! ${PY_TEST_WHITELIST} == *"${TEST_FILE_PATH}"* ]]; then
+    ((SKIP_COUNTER++))
+    echo "${PROG_STR} Skipping non-whitelisted test: ${TEST_FILE_PATH}"
+    continue
+  fi
+
+  # If the test is in the black list, skip it
+  if [[ ${PY_TEST_BLACKLIST} == *"${TEST_FILE_PATH}"* ]]; then
+    ((SKIP_COUNTER++))
+    echo "${PROG_STR} Skipping blacklisted test: ${TEST_FILE_PATH}"
+    continue
+  fi
+
+  # Copy to a separate directory to guard against the possibility of picking up
+  # modules in the source directory
+  cp ${TEST_FILE_PATH} ${PY_TEST_DIR}/
+
+  TEST_BASENAME=`basename "${TEST_FILE_PATH}"`
+
+  # Relative path of the test log. Use long path in case there are duplicate
+  # file names in the Python tests
+  TEST_LOG_REL="${PY_TEST_LOG_DIR_REL}/${TEST_FILE_PATH}.log"
+  mkdir -p `dirname ${TEST_LOG_REL}`  # Create directory for log
+
+  TEST_LOG=`abs_path ${TEST_LOG_REL}`  # Absolute path
+
+  # Before running the test, cd away from the Tensorflow source to
+  # avoid the possibility of picking up dependencies from the
+  # source directory
+  cd ${PY_TEST_DIR}
+  python ${PY_TEST_DIR}/${TEST_BASENAME} >${TEST_LOG} 2>&1
+
+  # Check for pass or failure status of the test outtput and exit
+  if [[ $? -eq 0 ]]; then
+    ((PASS_COUNTER++))
+
+    echo "${PROG_STR} Python test-on-install succeeded: ${TEST_FILE_PATH}"
```

Comment:
I'll change it to "PASSED", just to be shorter and syntactically consistent with "FAILED". Similarly, I'll changed "skipped" to "SKIPPED".